src/mono/wasm/Wasm.Build.Tests/TestMainJsProjectProvider.cs

```diff
@@ -0,0 +1,65 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using Xunit.Abstractions;
+
+namespace Wasm.Build.Tests;
+
+public class TestMainJsProjectProvider(string projectDir, ITestOutputHelper testOutput)
+                : ProjectProviderBase(projectDir, testOutput)
+{
+    // no fingerprinting
+    protected override IReadOnlyDictionary<string, bool> GetDotnetFilesKnownSet(RuntimeVariant runtimeType)
+        => new SortedDictionary<string, bool>()
```

Comment:
Could we merge `GetDotnetFilesKnownSet` and `GetDotNetFilesExpectedSet` and provide a collection of records (with properties)?

Reply:
And maybe a second thought. If I image that I add a new runtime file, I would need to find all places and add the file there. What I like about similar SDK tests, is that in such a case, I can just run all tests in mode "generate baselines" and than check the generated changes in git diff.

Reply:
The baselines stuff has been a major pain in sdk tests, and it's already well covered there. So I don't think we need to duplicate that here. One of the reasons for moving to this new way of testing for dotnet files was so the logic is all shared, and the difference is just the list of files. And the reason for difference list is that we still have different kinds of builds - blazor, templates, and test-main-js based tests, each having differences in their bundles. And these will gradually converge.

What I can do is try to ensure that a new runtime file would have to be added in only these lists. Does that make sense?

Reply:
I don't want to duplicate baseline tests from SDK, I'm just looking for a good way to discover all places for scenario when a new runtime file is added.

Reply:
> Could we merge GetDotnetFilesKnownSet and GetDotNetFilesExpectedSet and provide a collection of records (with properties)?

Do we need to method with two lists of files? The lists look identical. What am I missing?

Reply:
One is the list of all the known `dotnet.*` files to expect for a given project type, and whether to expect them to be fingerprinted. The other is for the actual list expected for individual tests. I can look at possibly improving this in the follow up PR.